first commit

burnsjared0415 opened the pull request:
Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
first commit

Signed-off-by: Jared Burns <burns.jared@gmail.com>

aaronlippold committed:
Removed repeated controls for now until we get things organized on the
controls using only_if and inputs.

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
fixed the senseitive example with only_if

Signed-off-by: Aaron Lippold <lippold@gmail.com>

burnsjared0415 committed:
Second Update to Controls

burnsjared0415 committed:
updated with only_if

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Moved Control From /controls/All_Systems to /controls

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Moved from /controls/All_Systems to /control

burnsjared0415 committed:
Moved /controls/All_Systems to /Control

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Moved Control from All_Systems

burnsjared0415 committed:
updated tag severity

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
updated tag severity

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
updated tag severity and also added only_if statement for controls that require it.

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Control to add If statement for VDI

Signed-off-by: Jared Burns <burns.jared@gmail.com>

aaronlippold commented on controls/All Systems/V-63323.rb:
```diff
@@ -64,5 +64,23 @@
 The TPM must be enabled in the firmware.
 Run \"tpm.msc\" for configuration options in Windows."
+  
+if is_domain == 'WORKGROUP'
```
Is `is_domain` an input? where is it defined?

aaronlippold commented on controls/All Systems/V-63323.rb:
```diff
@@ -64,5 +64,23 @@
+  desc 'This system is not joined to a domain, therfore this control is not appliable as it does not apply to standalone systems'
+end
+
+if is_domain != 'WORKGROUP'
```
you can use `unless` here

`unless input('is_domain') == WORKGROUP`

aaronlippold committed:
small updates

Signed-off-by: Aaron Lippold <lippold@gmail.com>

burnsjared0415 committed:
Updated Controls to fix some issues, one was for VDI

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Moved Controls

burnsjared0415 committed:
Moved everything out of All Systems to Controls

burnsjared0415 committed:
Updatad Controls to fix issues in if statements

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Controls to add a little more review of Control

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated If_only Statement and also updated Controls that had Cert checks

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated data for two controls V-63583 and V-63589

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Controls to remove \\ and replace with \ and also " and replace with '

Signed-off-by: Jared Burns <burns.jared@gmail.com>

aaronlippold committed:
First batch of fixes
- updated formatting and layout
- updated implementation of many of the NA controls
- fixed basic errors and missing end statments
- standardized the logic flow of many of the controls

Notes:

- We should add a ReleaseID propertiy to one of the windows resources
- Need to add the ability in only_if to update the impact
- Most of the unclassified controls could be implemented in one control that uses an input
  which is a hash perhaps?
- V-74725 - The NA case is not accounted for - and its somewhat insane need to talk to DISA
- v-74415 - does this mean it is na for 1507 and 1607 like the control after it ? etc?

Signed-off-by: Aaron Lippold <lippold@gmail.com>

burnsjared0415 committed:
Updated the Registry entries

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Fixed Registry Path

Signed-off-by: Jared Burns <burns.jared@gmail.com>

aaronlippold committed:
Merge pull request #3 from mitre/71769-94861

First batch of fixes

burnsjared0415 committed:
Updated Control V-63627 to add the property should not exist statement

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Controls to fix errors in describe

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
updated Control V-71769 for issue with end

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated V-71769 removed extra end

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Update

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated V-63627

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated V-63323

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Control V-63323

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated Control V-63323

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated V-63581

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Fixed Control V63581

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
updated control

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated describe.one block to add the should_not have property

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Merge branch 'jb-windows10-controls-enhancements' of https://github.com/mitre/microsoft-windows-10-stig-baseline into jb-windows10-controls-enhancements

burnsjared0415 committed:
Updated Describe Statements to move blocks around and cleaned up a if statement

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated control with a end statement

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Added End statements

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Was missing its

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Added dod to eca input

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated its to remove should be and make it should eq

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated its from should be to should cmp

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
changed back to should be

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Removed describe.one statment

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Idenified issues with values

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
remove " "

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated a couple issues with extra - and ""

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated from Be to CMP

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
change eq to be_in

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
updated to add be_in in place of cmp

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated the cmp to be_in and removed the describe.one on one of the controls to fix results.

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
removed extra end

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
fixed should statement to remove should be eq and changed to should be_in for one and eq for the second

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated its statement to not have include and have be_in

Signed-off-by: Jared Burns <burns.jared@gmail.com>

burnsjared0415 committed:
Updated statements

Signed-off-by: Jared Burns <burns.jared@gmail.com>

aaronlippold committed:
second round of updates
added inputs for password complexity and other useful params

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge branch 'jb-windows10-controls-enhancements' into batch2

aaronlippold committed:
Merge pull request #4 from mitre/batch2

second round of updates

aaronlippold committed:
other formatting changes

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge pull request #5 from mitre/71769-94861

other formatting changes

aaronlippold committed:
updated the check and fix tags to use sub-descriptions

Signed-off-by: Aaron Lippold <lippold@gmail.com>

aaronlippold committed:
Merge pull request #6 from mitre/check-fix-desc

updated the check and fix tags to use sub-descriptions
